chore(DAG-SINAN): proposal to split DAGs by disease

luabida opened the pull request:
# PROPOSTAL
 Change SINAN DAG by each disease instead of heaving one centralized DAG for the 36 diseases

## Motivations
- Some diseases take hours to fetch, having 35 running in the same DAG can take almost an entire day to fetch even with some concurrency, while some diseases take less than 10 minutes to insert the data.
![image](https://user-images.githubusercontent.com/82233055/218891242-6c00934f-a3e4-4f74-9c1f-a7b091b07097.png)
- Lack of schedule control. Having one DAG to all diseases, it is not possible to reduce or increase the schedule between each DAG run, creating an unnecessary bound between the diseases.
- Not scalable. I've notice the amount of diseases in SINAN may grow with time, which can increase even more the time the entire process takes to finish.
- High "breakability". The bound caused by the method `expand` (which loops between the diseases to fetch) is great because there is no need to consider how many diseases the DAG receives, but also creates a waterfall of errors if a disease stops working for any reason.
OBS:
Creating an task branch for each disease (as done in FOPH DAG) is unlikely to be an option due to the amount of diseases, it is very difficult to scroll down while DEBUGGING, because there will be a lot of tasks in the GUI.

## Cons
- Significantly increase of DAGs in the GUI. It is also not possible to use non-ASCII characters in the DAG's name. A good workaround is to define properly tags for each DAG created so it can be easily seached. The result follows:
![image](https://user-images.githubusercontent.com/82233055/218893646-f728fd32-3937-4ca6-b33d-998be0d2475d.png)

----
depends on https://github.com/thegraphnetwork/epigraphhub_py/pull/211

luabida commented:
![image](https://user-images.githubusercontent.com/82233055/218895233-73f05cc4-3bfe-4412-9760-92dd4e6930d2.png)

luabida commented:
> you can also pass parameters to each run

Ideally, there would be a function like `generate_dag(disease, schedule, start)` that would be responsible for creating DAGs with no need of repeating them as I did.
I've also tried using jinja2 syntax:
``` python
with DAG(
    dag_id=DISEASES["{{ disease }}"], 
    default_args=DEFAULT_ARGS,
    tags=["{{ disease }}", 'SINAN'],
    start_date= "{{ start }}",
    catchup=False,
    schedule_interval="{{ schedule }}",
) as "{{ disease_code }}":
    task_flow_for("{{ disease }}")
```

luabida commented:
> And please remember to use semantic table names as per Epigraphhub documentation.

There are some long names within diseases list, should I use the extended name as well?
e.g. `brazil.sinan_contact_communicable_disease_m`

luabida commented:
Following [this tutorial](https://docs.astronomer.io/learn/dynamically-generating-dags), I was able to dynamic generate DAGs using the diseases dictionary from SINAN using this `globals()[dag_id]` syntax. I've also added a `SINAN_` prefix to DAG's names because I think it's more explicit. The results are:
![image](https://user-images.githubusercontent.com/82233055/219135167-6d24621a-78cb-4379-83a6-a20bbc72211d.png)
![image](https://user-images.githubusercontent.com/82233055/219136083-8d627976-8c50-4437-be8b-0e5eef40e77a.png)

luabida commented:
Added a cleaner visualization on diagnosis task, located on XCom tab:
![image](https://user-images.githubusercontent.com/82233055/219161149-49462bdf-9078-47ae-8cec-d67b5986096e.png)

luabida commented:
I've encountered this error while trying to insert `Sifilis em Gestante` data. Have you ever encountered a error like that? Could it be a PySUS encoding error?

```
Traceback (most recent call last):
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1471, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1618, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1679, in _execute_task
    result = execute_callable(context=context)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/decorators/base.py", line 179, in execute
    return_value = super().execute(context)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/operators/python.py", line 171, in execute
    return_value = self.execute_callable()
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/operators/python.py", line 189, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/brasil/sinan.py", line 77, in upload
    raise e
  File "/opt/airflow/dags/brasil/sinan.py", line 74, in upload
    loading.upload(parquet_dirs)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/epigraphhub/data/brasil/sinan/loading.py", line 66, in upload
    raise e
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/epigraphhub/data/brasil/sinan/loading.py", line 51, in upload
    upsert(
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/core.py", line 302, in upsert
    executor.execute(connectable=con, if_row_exists=if_row_exists, chunksize=chunksize)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/executor.py", line 87, in execute
    pse.upsert(if_row_exists=if_row_exists, chunksize=chunksize)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/engine.py", line 551, in upsert
    upq.execute(db_type=self._db_type, values=chunk, if_row_exists=if_row_exists)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/upsert_query.py", line 231, in execute
    return self.connection.execute(query)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1380, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
    ret = self._execute_context(
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2128, in _handle_dbapi_exception
    util.raise_(exc_info[1], with_traceback=exc_info[2])
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
ValueError: A string literal cannot contain NUL (0x00) characters.
```

luabida commented:
results locally:

![image](https://user-images.githubusercontent.com/82233055/219451696-3ae7b322-8a1f-400f-af69-4a1fdafd1429.png)

![image](https://user-images.githubusercontent.com/82233055/219451771-e95f914c-0f39-499e-be85-27eb2ae7883e.png)

luabida commented:
Issue with SIFG fixed:
![image](https://user-images.githubusercontent.com/82233055/219453750-4a22d20d-b68e-4f12-9758-b9f3e5a0c905.png)

fccoelho commented:
> 

As for most tables the disease name is just one or two words. we must stick to the name to maintain the semantic naming criterion.

fccoelho commented:
> Edit: Fixed in [2f84a99](https://github.com/thegraphnetwork/epigraphhub_py/pull/211/commits/2f84a9925e5f0f3f5de07247f73f21a962dd0978)
> 
> I've encountered this error while trying to insert `Sifilis em Gestante` data. Have you ever encountered a error like that? Could it be a PySUS encoding error?
> 
> ```
> Traceback (most recent call last):
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1471, in _run_raw_task
>     self._execute_task_with_callbacks(context, test_mode)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1618, in _execute_task_with_callbacks
>     result = self._execute_task(context, task_orig)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1679, in _execute_task
>     result = execute_callable(context=context)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/decorators/base.py", line 179, in execute
>     return_value = super().execute(context)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/operators/python.py", line 171, in execute
>     return_value = self.execute_callable()
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/airflow/operators/python.py", line 189, in execute_callable
>     return self.python_callable(*self.op_args, **self.op_kwargs)
>   File "/opt/airflow/dags/brasil/sinan.py", line 77, in upload
>     raise e
>   File "/opt/airflow/dags/brasil/sinan.py", line 74, in upload
>     loading.upload(parquet_dirs)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/epigraphhub/data/brasil/sinan/loading.py", line 66, in upload
>     raise e
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/epigraphhub/data/brasil/sinan/loading.py", line 51, in upload
>     upsert(
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/core.py", line 302, in upsert
>     executor.execute(connectable=con, if_row_exists=if_row_exists, chunksize=chunksize)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/executor.py", line 87, in execute
>     pse.upsert(if_row_exists=if_row_exists, chunksize=chunksize)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/engine.py", line 551, in upsert
>     upq.execute(db_type=self._db_type, values=chunk, if_row_exists=if_row_exists)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/pangres/upsert_query.py", line 231, in execute
>     return self.connection.execute(query)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1380, in execute
>     return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
>     return connection._execute_clauseelement(
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
>     ret = self._execute_context(
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
>     self._handle_dbapi_exception(
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2128, in _handle_dbapi_exception
>     util.raise_(exc_info[1], with_traceback=exc_info[2])
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
>     raise exception
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
>     self.dialect.do_execute(
>   File "/opt/conda/envs/epigraphhub/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
>     cursor.execute(statement, parameters)
> ValueError: A string literal cannot contain NUL (0x00) characters.
> ```

I don't remember facing such an issue. It seem that some invalid unicode characters in the SINAN data, in this case it will help to log the offending record so that we can inspect  the problem and find a work around.

fccoelho commented:
> Issue with SIFG fixed: ![image](https://user-images.githubusercontent.com/82233055/219453750-4a22d20d-b68e-4f12-9758-b9f3e5a0c905.png)

Did you find out what was causing the problem?

luabida commented:
> I don't remember facing such an issue. It seem that some invalid unicode characters in the SINAN data, in this case it will help to log the offending record so that we can inspect the problem and find a work around.

It took me a while to figure out that the file `/tmp/pysus/SIFGBR14.parquet` had some NUL ('\x00') values in it, I couldn't find another workaround but trimming every parquet chunk (which pandas does pretty fast with `apply`), [here](https://github.com/thegraphnetwork/epigraphhub_py/pull/211/files#diff-9b95529118af03ddb605f0deff147ec424758bd85833a95679614dabd04b4143R74)

luabida commented:
```
sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type bigint: " "
LINE 1: ...                             ', '        ', 1, 1, ' ', '    ...
                                                             ^

[SQL: INSERT INTO brasil.sinan_tetano_neonatal_m (index, tp_not, id_agravo, dt_notific, sem_not, nu_ano, sg_uf_not, id_municip, id_regiona, id_unidade, dt_sin_pri, sem_pri, dt_nasc, nu_idade_n, cs_raca, cs_escol_n, sg_uf, id_mn_resi, id_rg_resi, id_pais, dt_invest, id_ocupa_n, num_con_n, cs_vacteta, dt_1_dose, dt_2_dose, dt_3_dose, dt_reforco, idade_mae, nu_gesta, escolmae_n, cs_nascido, no_ouparto, cs_atend_n, no_atenout, cs_sugou, cs_mamar, cs_trismo, cs_nuca, cs_choro, cs_riso, cs_abdomen, cs_inf_cot, cs_opistot, cs_membros, cs_crise, cs_outros, nm_out_sin, dt_trismo, cs_origem, cs_hospita, cs_cobertu, no_coboutr, cs_vacinac, cs_cadastr, cs_divulga, cs_buscaat, cs_orienta, cs_analise, cs_outras, no_outras, classi_fin, ds_inf_loc, ds_inf_out, tpautocto, coufinf, copaisinf, comuninf, counidinf, evolucao, dt_obito, dt_encerra, dt_digita, dt_transdm) VALUES (%(index_m0)s, %(tp_not_m0)s, %(id_agravo_m0)s, %(dt_notific_m0)s, %(sem_not_m0)s, %(nu_ano_m0)s, %(sg_uf_not_m0)s, %(id_municip_m0)s, %(id_regiona_m0)s, %(id_unidade_m0)s, %(dt_sin_pri_m0)s, %(sem_pri_m0)s, %(dt_nasc_m0)s, %(nu_idade_n_m0)s, %(cs_raca_m0)s, %(cs_escol_n_m0)s, %(sg_uf_m0)s, %(id_mn_resi_m0)s, %(id_rg_resi_m0)s, %(id_pais_m0)s, %(dt_invest_m0)s, %(id_ocupa_n_m0)s, %(num_con_n_m0)s, %(cs_vacteta_m0)s, %(dt_1_dose_m0)s, %(dt_2_dose_m0)s, %(dt_3_dose_m0)s, %(dt_reforco_m0)s, %(idade_mae_m0)s, %(nu_gesta_m0)s, %(escolmae_n_m0)s, %(cs_nascido_m0)s, %(no_ouparto_m0)s, %(cs_atend_n_m0)s, %(no_atenout_m0)s, %(cs_sugou_m0)s, %(cs_mamar_m0)s, %(cs_trismo_m0)s, %(cs_nuca_m0)s, %(cs_choro_m0)s, %(cs_riso_m0)s, %(cs_abdomen_m0)s, %(cs_inf_cot_m0)s, %(cs_opistot_m0)s, %(cs_membros_m0)s, %(cs_crise_m0)s, %(cs_outros_m0)s, %(nm_out_sin_m0)s, %(dt_trismo_m0)s, %(cs_origem_m0)s, %(cs_hospita_m0)s, %(cs_cobertu_m0)s, %(no_coboutr_m0)s, %(cs_vacinac_m0)s, %(cs_cadastr_m0)s, %(cs_divulga_m0)s, %(cs_buscaat_m0)s, %(cs_orienta_m0)s, %(cs_analise_m0)s, %(cs_outras_m0)s, %(no_outras_m0)s, %(classi_fin_m0)s, %(ds_inf_loc_m0)s, %(ds_inf_out_m0)s, %(tpautocto_m0)s, %(coufinf_m0)s, %(copaisinf_m0)s, %(comuninf_m0)s, %(counidinf_m0)s, %(evolucao_m0)s, %(dt_obito_m0)s, %(dt_encerra_m0)s, %(dt_digita_m0)s, %(dt_transdm_m0)s) ON CONFLICT (index) DO UPDATE SET tp_not = excluded.tp_not, id_agravo = excluded.id_agravo, dt_notific = excluded.dt_notific, sem_not = excluded.sem_not, nu_ano = excluded.nu_ano, sg_uf_not = excluded.sg_uf_not, id_municip = excluded.id_municip, id_regiona = excluded.id_regiona, id_unidade = excluded.id_unidade, dt_sin_pri = excluded.dt_sin_pri, sem_pri = excluded.sem_pri, dt_nasc = excluded.dt_nasc, nu_idade_n = excluded.nu_idade_n, cs_raca = excluded.cs_raca, cs_escol_n = excluded.cs_escol_n, sg_uf = excluded.sg_uf, id_mn_resi = excluded.id_mn_resi, id_rg_resi = excluded.id_rg_resi, id_pais = excluded.id_pais, dt_invest = excluded.dt_invest, id_ocupa_n = excluded.id_ocupa_n, num_con_n = excluded.num_con_n, cs_vacteta = excluded.cs_vacteta, dt_1_dose = excluded.dt_1_dose, dt_2_dose = excluded.dt_2_dose, dt_3_dose = excluded.dt_3_dose, dt_reforco = excluded.dt_reforco, idade_mae = excluded.idade_mae, nu_gesta = excluded.nu_gesta, escolmae_n = excluded.escolmae_n, cs_nascido = excluded.cs_nascido, no_ouparto = excluded.no_ouparto, cs_atend_n = excluded.cs_atend_n, no_atenout = excluded.no_atenout, cs_sugou = excluded.cs_sugou, cs_mamar = excluded.cs_mamar, cs_trismo = excluded.cs_trismo, cs_nuca = excluded.cs_nuca, cs_choro = excluded.cs_choro, cs_riso = excluded.cs_riso, cs_abdomen = excluded.cs_abdomen, cs_inf_cot = excluded.cs_inf_cot, cs_opistot = excluded.cs_opistot, cs_membros = excluded.cs_membros, cs_crise = excluded.cs_crise, cs_outros = excluded.cs_outros, nm_out_sin = excluded.nm_out_sin, dt_trismo = excluded.dt_trismo, cs_origem = excluded.cs_origem, cs_hospita = excluded.cs_hospita, cs_cobertu = excluded.cs_cobertu, no_coboutr = excluded.no_coboutr, cs_vacinac = excluded.cs_vacinac, cs_cadastr = excluded.cs_cadastr, cs_divulga = excluded.cs_divulga, cs_buscaat = excluded.cs_buscaat, cs_orienta = excluded.cs_orienta, cs_analise = excluded.cs_analise, cs_outras = excluded.cs_outras, no_outras = excluded.no_outras, classi_fin = excluded.classi_fin, ds_inf_loc = excluded.ds_inf_loc, ds_inf_out = excluded.ds_inf_out, tpautocto = excluded.tpautocto, coufinf = excluded.coufinf, copaisinf = excluded.copaisinf, comuninf = excluded.comuninf, counidinf = excluded.counidinf, evolucao = excluded.evolucao, dt_obito = excluded.dt_obito, dt_encerra = excluded.dt_encerra, dt_digita = excluded.dt_digita, dt_transdm = excluded.dt_transdm]
[parameters: {'index_m0': 0, 'tp_not_m0': 2, 'id_agravo_m0': 'A33 ', 'dt_notific_m0': 20171113, 'sem_not_m0': 201746, 'nu_ano_m0': 2017, 'sg_uf_not_m0': '25', 'id_municip_m0': 251080, 'id_regiona_m0': '1422    ', 'id_unidade_m0': 2605414, 'dt_sin_pri_m0': '20171112', 'sem_pri_m0': '201746', 'dt_nasc_m0': '20171108', 'nu_idade_n_m0': '2004', 'cs_raca_m0': '4', 'cs_escol_n_m0': '10', 'sg_uf_m0': '25', 'id_mn_resi_m0': '251620', 'id_rg_resi_m0': '1426    ', 'id_pais_m0': '1   ', 'dt_invest_m0': 20171113, 'id_ocupa_n_m0': 999992, 'num_con_n_m0': 1, 'cs_vacteta_m0': 1, 'dt_1_dose_m0': 20170927, 'dt_2_dose_m0': '        ', 'dt_3_dose_m0': '        ', 'dt_reforco_m0': '        ', 'idade_mae_m0': 41, 'nu_gesta_m0': 2, 'escolmae_n_m0': 3, 'cs_nascido_m0': 1, 'no_ouparto_m0': '                              ', 'cs_atend_n_m0': 1, 'no_atenout_m0': '                              ', 'cs_sugou_m0': 2, 'cs_mamar_m0': 2, 'cs_trismo_m0': 2, 'cs_nuca_m0': 1, 'cs_choro_m0': 2, 'cs_riso_m0': 1, 'cs_abdomen_m0': 2, 'cs_inf_cot_m0': 1, 'cs_opistot_m0': 1, 'cs_membros_m0': 1, 'cs_crise_m0': 1, 'cs_outros_m0': 2, 'nm_out_sin_m0': '                              ', 'dt_trismo_m0': '        ', 'cs_origem_m0': 1, 'cs_hospita_m0': 1, 'cs_cobertu_m0': ' ', 'no_coboutr_m0': '                              ', 'cs_vacinac_m0': 1, 'cs_cadastr_m0': 2, 'cs_divulga_m0': 1, 'cs_buscaat_m0': 1, 'cs_orienta_m0': 2, 'cs_analise_m0': 2, 'cs_outras_m0': 2, 'no_outras_m0': '                              ', 'classi_fin_m0': 2, 'ds_inf_loc_m0': 1, 'ds_inf_out_m0': '                              ', 'tpautocto_m0': 3, 'coufinf_m0': '  ', 'copaisinf_m0': 0, 'comuninf_m0': '      ', 'counidinf_m0': '       ', 'evolucao_m0': 3, 'dt_obito_m0': 20171113, 'dt_encerra_m0': 20171114, 'dt_digita_m0': 20171114, 'dt_transdm_m0': '        '}]
(Background on this error at: https://sqlalche.me/e/14/9h9h)
[2023-02-27, 17:41:51 UTC] {{taskinstance.py:1415}} INFO - Marking task as UP_FOR_RETRY. dag_id=SINAN_TETN, task_id=upload, execution_date=20230227T224141, start_date=20230227T224149, end_date=20230227T224151
[2023-02-27, 17:41:51 UTC] {{standard_task_runner.py:92}} ERROR - Failed to execute job 21 for task upload ((psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type bigint: " "
LINE 1: ...                             ', '        ', 1, 1, ' ', '    ...
                                                             ^
                                                             
                                                           ```

luabida commented:
## Overview about changes made:
### [PySUS](https://github.com/AlertaDengue/PySUS/pull/117)
- Added new SINAN classes:
I've chose to add new classes to download and read parquet files into pandas dataframe trying to not change the legacy methods. The new classes are available [here](https://github.com/AlertaDengue/PySUS/pull/117/files#diff-90cab74b31593b4252cdc24e6758c7b281e42d3b4474b9f188597b49591cc9fb). The reading method now cast columns according to [this](https://github.com/AlertaDengue/PySUS/pull/117/files#diff-0002fc717305a0e28a78aa038e794a8d2b745c1506515e57d28ef2e20959258c) but not every variable has been found at the metadata PDFs, which were kept as type `object`
- Added a metadata method:
This method reads from the converted PDFs into csv.tar.gz, usage: `SINAN.metadata_df(disease)`. Not every disease was found in PDFs.
- All columns use [utf-8 decode](https://github.com/AlertaDengue/PySUS/pull/117/files#diff-90cab74b31593b4252cdc24e6758c7b281e42d3b4474b9f188597b49591cc9fbR229)
- Numeric columns [remove every non-numeric number](https://github.com/AlertaDengue/PySUS/pull/117/files#diff-90cab74b31593b4252cdc24e6758c7b281e42d3b4474b9f188597b49591cc9fbR244) 
**WARNING**: The cast process has the possibility of loosing data, but the only workaround to properly insert correct data types would be seeing case by case  

### [epigraphhub_py](https://github.com/thegraphnetwork/epigraphhub_py/pull/211)
- Tables now follow the [EGH tables naming convention](https://epigraphhub.readthedocs.io/en/latest/instruction_name_tables.html#all-names-schema-name-table-name-and-column-names)

### [EpiGraphHub](https://github.com/thegraphnetwork/EpiGraphHub/pull/170)
- [Diseases DAG split into a DAG to each disease](https://github.com/thegraphnetwork/EpiGraphHub/pull/170/files#diff-d5d15222274f0bb0605fc5e65ff7ca17281a4fe6716f4029f20000a041b84b63R183)
- [Metadata DAG added](https://github.com/thegraphnetwork/EpiGraphHub/pull/170/files#diff-d5d15222274f0bb0605fc5e65ff7ca17281a4fe6716f4029f20000a041b84b63R147)

luabida commented:
Depends on https://github.com/AlertaDengue/PySUS/pull/117 and https://github.com/thegraphnetwork/epigraphhub_py/pull/211 merge & release

luabida commented:
@fccoelho @xmnlab ready to review
